third_party/flower/src/brave/flwr/serde.h

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_THIRD_PARTY_FLOWER_SRC_BRAVE_FLWR_SERDE_H_
+#define BRAVE_THIRD_PARTY_FLOWER_SRC_BRAVE_FLWR_SERDE_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "brave/third_party/flower/src/proto/flwr/proto/fleet.pb.h"
+#include "brave/third_party/flower/src/proto/flwr/proto/transport.pb.h"
+
+class ScalarValue {
+ public:
+  // Getters
+  std::optional<bool> GetBool() { return bool_value_; }
+  std::optional<std::string> GetBytes() { return bytes_value_; }
+  std::optional<double> GetDouble() { return double_value_; }
+  std::optional<int> GetInt() { return integer_value_; }
+  std::optional<std::string> GetString() { return string_value_; }
+
+  // Setters
+  void SetBool(bool b) { this->bool_value_ = b; }
+  void SetBytes(std::string bytes) { this->bytes_value_ = bytes; }
+  void SetDouble(double d) { this->double_value_ = d; }
+  void SetInt(int i) { this->integer_value_ = i; }
+  void SetString(std::string string) { this->string_value_ = string; }
+
+ private:
+  std::optional<bool> bool_value_ = std::nullopt;
```

Comment:
no need to `= std::nullopt`. Also we should not be using std::optional, see `base::optional`

Reply:
do you mean `absl::optional`?

Reply:
Ah yes absl::optional